src/gui/transferlistwidget.cpp

```diff
@@ -1327,7 +1327,7 @@ void TransferListWidget::applyFilter(const QString &name, const TransferListMode
 {
     m_sortFilterModel->setFilterKeyColumn(type);
     const QString pattern = (Preferences::instance()->getRegexAsFilteringPatternForTransferList()
-                ? name : Utils::String::wildcardToRegexPattern(name));
+                ? name : QRegularExpression::escape(Utils::String::wildcardToRegexPattern(name)));
```

Comment:
IIRC, this disables all "wildcard" capabilities that are supposed to be provided for user.

Reply:
Sorry, I didn't take wildcard patterns into consideration. In wildcard patterns, it's not possible to differentiate between symbols like "[", "]", "*", etc. Should add a fuzzy search feature? Similar to the regular expressions mode in the menu.

Reply:
https://github.com/qbittorrent/qBittorrent/issues/19623#issuecomment-1731040326